luminoth/tools/dataset/readers/object_detection/imagenet.py

```diff
@@ -0,0 +1,209 @@
+import json
+import os
+import random
+import tensorflow as tf
+
+from PIL import Image
+
+from luminoth.tools.dataset.readers import InvalidDataDirectory
+from luminoth.tools.dataset.readers.object_detection import (
+    ObjectDetectionReader
+)
+from luminoth.utils.dataset import read_xml, read_image
+
+WNIDS_FILE = 'data/imagenet_wnids.json'
+
+
+class ImageNetReader(ObjectDetectionReader):
+    def __init__(self, data_dir, split, only_filename=None,
+                 limit_examples=None, limit_classes=None, seed=None, **kwargs):
+        super(ImageNetReader, self).__init__()
+        self._split = split
+        self._data_dir = data_dir
+        self._imagesets_path = os.path.join(self._data_dir, 'ImageSets', 'DET')
+        self._images_path = os.path.join(self._data_dir, 'Data', 'DET',)
+        self._annotations_path = os.path.join(
+            self._data_dir, 'Annotations', 'DET'
+        )
+
+        self._only_filename = only_filename
+        self._limit_examples = limit_examples
+        self._limit_classes = limit_classes
```

Comment:
Maybe also accept that `_limit_classes` can be a list of classes which we want to keep? Instead of a simple number.

Used together with `_limit_examples` this could be powerful for reproducibility while debugging.